src/main/java/org/opentestsystem/delivery/logging/EventLogger.java

```diff
@@ -7,31 +7,149 @@
+    LOGIN,
+    LOGOUT,
+    UNKNOWN,
+  }
+
+  public enum EventData implements IEventData {
+    APP,
+    ELAPSED_TIME,
+    CHECKPOINT,
+    SESSION_ID,
+    SUB_EVENT,
+  }
+
+  /**
+   * Log event occurrences to centralized logging.
+   * <p>
+   * These events are intended to be sent to a logstash server and searchable using elastic search.
+   *
+   * @param app  name of the app writing the log
+   * @param logEvent  name of the event occurring
+   * @param checkpoint  an optional checkpoint name
+   * @param sessionId  an optional unique identifier common to a set of log entries (generally a request's session ID)
+   * @param message  an optional free-format message string to add to the log entry
+   * @param data optional searchable event data to add to the recorded event
+   */
+  public static void info(final String app, final ILogEvent logEvent, final String checkpoint, final String sessionId,
+                          final String message, final Map<IEventData, Object> data) {
+    log(formatMessage(logEvent, message), getFieldMap(app, logEvent, checkpoint, sessionId, data));
+  }
+
+  /**
+   * Log event errors to centralized logging.
+   * <p>
+   * These events are intended to be sent to a logstash server and searchable using elastic search.
+   *
+   * @param app  name of the app writing the log
+   * @param logEvent  name of the event occurring
+   * @param checkpoint  an optional checkpoint name
+   * @param sessionId  an optional unique identifier common to a set of log entries (generally a request's session ID)
+   * @param message  an optional free-format message string to add to the log entry
+   * @param data optional searchable event data to add to the recorded event
+   */
+  public static void error(final String app, final ILogEvent logEvent, final String checkpoint, final String sessionId,
+                           String message, final Map<IEventData, Object> data, final Exception e) {
+    error(formatMessage(logEvent, message), getFieldMap(app, logEvent, checkpoint, sessionId, data), e);
+  }
+
+  private static void log(final String event, final Map<String, Object> fields) {
+    try {
+      logger.info(marker, format("EVENT:%s JSON:({\"event_data\": %s})", event, writer.writeValueAsString(fields)));
+    } catch (Exception e) {
+      logger.warn(marker, format("exception occurred while logging event: %s", event), e);
+    }
+  }
+
+  private static void error(final String event, final Map<String, Object> fields, final Exception e) {
+    try {
+      logger.error(marker, format("EVENT:%s JSON:({\"event_data\": %s})", event, writer.writeValueAsString(fields)),
+        e);
+    } catch (Exception inner) {
+      logger.error(marker, format("exception occurred while error event: %s", event), inner);
+    }
+  }
+
+  private static void addTimerMetrics(final ILogEvent event, final String checkpoint, Map<IEventData, Object> fields) {
+    Long eventTime = eventTimes.get(event);
+    switch (checkpoint) {
```

Comment:
Ideally, yes, but I wanted checkpoint to be free-form text so it's easy to create arbitrary checkpoints from the application code (there's an example with EXAM_PAUSE in student app). 
Also the way I did the enum inheritance breaks switch statements (it was a switch on an enum before I made it a string and added that inheritance). I could change it to an if-else and then match against enum.name(), which would get rid of the case statement magic values.